src/index.ts

```diff
@@ -459,60 +456,103 @@ async function explicitInit(config: IClientConfig): Promise<EppoClient> {
         if (cacheIsExpired && !config.useExpiredCache) {
           applicationLogger.warn('Eppo SDK set not to use expired cached configuration');
           initFromConfigStoreError = new Error('Configuration store was expired');
-          return '';
+          return ConfigLoaderStatus.DID_NOT_PRODUCE;
+        } else if (cacheIsExpired && config.useExpiredCache) {
+          applicationLogger.warn('Eppo SDK config.useExpiredCache is true; using expired cache');
         }
-        return 'config store';
+
+        return ConfigLoaderStatus.COMPLETED;
       })
       .catch((e) => {
         applicationLogger.warn(
           'Eppo SDK encountered an error initializing from the configuration store',
           e,
         );
         initFromConfigStoreError = e;
+        return ConfigLoaderStatus.FAILED;
+      })
+      .then((status) => {
+        return [ConfigSource.CONFIG_STORE, status];
       });
-    const attemptInitFromFetch = instance
+
+    const attemptInitFromFetch: ConfigurationLoadAttempt = instance
       .fetchFlagConfigurations()
-      .then(() => 'fetch')
+      .then(() => {
+        console.log(instance.getFlagConfigurations());
+        if (configurationStore.isInitialized()) {
+          // If fetch has won the race and the configStore is initialized, that means we have a successful load and the
+          // fetched data was used to init.
+          return ConfigLoaderStatus.COMPLETED;
+        } else {
+          // If the config store is not initialized and the fetch says it is done, that means the fetch did not set any
+          // values into the config store (ex:the cache is not expired or invalid config data was received).
+          // It's important not to set an error here as this state does not mean that an error was encountered.
+          return ConfigLoaderStatus.DID_NOT_PRODUCE;
+        }
+      })
       .catch((e) => {
         applicationLogger.warn('Eppo SDK encountered an error initializing from fetching', e);
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         initFromFetchError = e;
+        return ConfigLoaderStatus.FAILED;
+      })
+      .then((status) => {
+        return [ConfigSource.FETCH, status];
       });
 
-    let initializationSource = await Promise.race([
-      attemptInitFromConfigStore,
-      attemptInitFromFetch,
-    ]);
+    const racers: ConfigurationLoadAttempt[] = [attemptInitFromConfigStore];
 
-    if (!initializationSource) {
-      // First attempt failed, but we have a second at bat that will be executed in the scope of the top-level try-catch
-      if (!initFromConfigStoreError) {
-        initializationSource = await attemptInitFromConfigStore;
-      } else {
-        initializationSource = await attemptInitFromFetch;
-      }
+    // attemptInitFromFetch only has side effects if `skipInitialRequest` is false.
+    if (!config.skipInitialRequest) {
+      racers.push(attemptInitFromFetch);
```

Comment:
No more slightly delayed start for fetch? How do we prevent fetch from trying if the configuration store inits very quickly? Or is this the new paradigm where by the time we get here everything should be initialized? (sorry--hard for me to keep track of all the conversation 😛 )

Reply:
The poller callback does a check to see if the config is expired before making the actual network call. This happens in the common lib. Added some more comments here and elsewhere.